src/playwright/playwright-controller.ts

```diff
@@ -27,14 +28,41 @@ export class PlaywrightController extends BrowserController<BrowserType, Paramet
             throw new Error('A new page can be created with provided context only when using incognito pages.');
         }
 
-        const page = await this.browser.newPage(contextOptions);
-        tryCancel();
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        let close = async () => {};
+        if (contextOptions?.proxy) {
+            const [anonymizedProxyUrl, closeProxy] = await anonymizeProxySugar(
+                contextOptions.proxy.server,
+                contextOptions.proxy.username,
+                contextOptions.proxy.password,
+            );
 
-        page.once('close', () => {
-            this.activePages--;
-        });
+            if (anonymizedProxyUrl) {
+                contextOptions.proxy = {
+                    server: anonymizedProxyUrl,
+                    bypass: contextOptions.proxy.bypass,
+                };
+            }
 
-        return page;
+            close = closeProxy;
+        }
+
+        try {
+            const page = await this.browser.newPage(contextOptions);
+            tryCancel();
+
+            page.once('close', async () => {
+                this.activePages--;
+
+                await close();
```

Comment:
This is quite unsafe because the page may open popups and get closed after opening them.

Reply:
plus this can crash the entire process if `close()` throws for some unforeseeable reason

Reply:
@pocesar

> plus this can crash the entire process if `close()` throws for some unforeseeable reason

Yes, that's what we want. If it failed to close the server (which should never happen) then it leaks resources which means we want to crash.

Reply:
> This is quite unsafe because the page may open popups and get closed after opening them.

Yeah, I see. But at the same time, closing those extra pages as well might be unexpected for users. Let's live with it and see if it causes any problems. The browsers have a hard kill interval anyway.